sirstrlcat.h

```diff
@@ -0,0 +1,67 @@
+/* SPDX-License-Identifier: ISC and MIT */
+
+/*
+ * Header-only strlcat implementation
+ *
+ * Derived from: $OpenBSD: strlcat.c,v 1.19 2019/01/25 00:19:25 millert Exp $
+ *
+ * Copyright (c) 1998, 2015 Todd C. Miller <millert@openbsd.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#undef strlcat
+
+/*
+ * Appends src to string dst of size dsize (unlike strncat, dsize is the
+ * full size of dst, not space left).  At most dsize-1 characters
+ * will be copied.  Always NUL terminates (unless dsize <= strlen(dst)).
+ * Returns strlen(src) + MIN(dsize, strlen(initial dst)).
+ * If retval >= dsize, truncation occurred.
+ */
+
+static inline size_t
+_sir_strlcat(char *dst, const char *src, size_t dsize)
+{
+  const char * odst  =   dst;
+  const char * osrc  =   src;
+  size_t          n  = dsize;
+  size_t       dlen;
+
+  /* Find the end of dst and adjust bytes left but don't go past end. */
+
+  while (n-- != 0 && *dst != '\0')
+    { dst++; }
+
+  dlen  =   dst - odst;
+  n     = dsize - dlen;
```

Comment:
This is good, but I will end up reformatting it to better match the style of everything else we have

Reply:
I'm certainly not married to the formatting in any way. 👍🏻